staff/staff_views_helper.py

```diff
@@ -208,3 +213,150 @@ def add_volunteers_to_volunteer_dict(classroom):
+    remove_enrollment(vol_contact, class_offering)
+
+
+def remove_student(request):
+    student_contact = get_contact_by_user_id(request.POST["former_student"])
+    class_offering = get_class_offering_by_id(request.POST["course_id"])
+    classroom = Classroom.objects.get(id=request.POST["course_id"])
+    classroom.students.remove(request.POST["former_student"])
+    remove_enrollment(student_contact, class_offering)
+
+
+def add_volunteers(request):
+    form = AddVolunteersForm(request.POST)
+    class_offering = get_class_offering_by_id(request.POST["course_id"])
+    classroom = Classroom.objects.get(id=request.POST["course_id"])
+    if form.is_valid():
+        for volunteer in form.cleaned_data.get("volunteers"):
+            vol_contact = get_contact_by_user_id(volunteer)
+            ClassEnrollment.objects.create(
+                created_by=class_offering.created_by,
+                contact=vol_contact,
+                status="Enrolled",
+                class_offering=class_offering,
+            )
+            classroom.volunteers.add(volunteer)
+
+
+def add_students(request):
+    form = AddStudentsForm(request.POST)
+    class_offering = get_class_offering_by_id(request.POST["course_id"])
+    classroom = Classroom.objects.get(id=request.POST["course_id"])
+    if form.is_valid():
+        for student in form.cleaned_data.get("students"):
+            student_contact = get_contact_by_user_id(student)
+            ClassEnrollment.objects.create(
+                created_by=class_offering.created_by,
+                contact=student_contact,
+                status="Enrolled",
+                class_offering=class_offering,
+            )
+            classroom.students.add(student)
+
+
+def remove_enrollment(contact, class_offering):
+    old_enrollment = ClassEnrollment.objects.get(
+        contact=contact, status="Enrolled", class_offering=class_offering
+    )
+    created_by = old_enrollment.created_by
+    old_enrollment.delete()
+    return created_by
+
+
+def get_contact_by_user_id(id):
+    return Contact.objects.get(
+        client_id=UserProfile.objects.get(user_id=id).salesforce_id
+    )
+
+
+def get_class_offering_by_id(id):
+    course_name = get_course_name_by_id(id)
+    return ClassOffering.objects.get(name=course_name)
+
+
+def get_course_name_by_id(id):
+    return Classroom.objects.get(id=id).course
+
+
+def sync_attendance_with_salesforce_class_offerings():
+    """
+    This method is fluid; in its current state it will duplicate entries
+    in your postgres templates database if you've already got templates data!
+    Only use if you don't have templates data but do have Classroom
+    data.
+    """
+    for classoffering in ClassOffering.objects.all():
+        dates = class_offering_meeting_dates(classoffering)
+        classroom = Classroom.objects.get(course=classoffering.name)
+        for day in dates:
+            session = Session.objects.create()
```

Comment:
It seems odd that the Session has so little information associated with it (no day or class or class offering) yet it's uniquely determined by the combination of those things in this implementation.

Reply:
I originally had "date" in the Session model, but switched it to the Attendance model, and each entry in the Attendance model has a Session Id associated with it. The thought was that perhaps an Absent marking in the Attendance model could trigger an email with a file from that day's Session to be auto-generated and sent to the absent student (this comes from my experience of it being a pain to catch students up who are absent and then email me "What did I miss?"  

It wouldn't hurt to put "date" in both Session and Attendance though.  Is that what you're thinking I should do?

Reply:
Session probably should have a foreign key to classroom and the date, so you can find sessions for a class without looking at attendance records. We could populate the description from a template that uses the name of the Classroom and/or ClassOffering.

I think it's fine to have the date be denormalized to both of these tables given how we're querying for it.